src/openms/include/OpenMS/FILTERING/CALIBRATION/TrafoModel.h

```diff
@@ -0,0 +1,315 @@
+    The models are build based on this input.
+
+    Outlier detection before model building via the RANSAC algorithm is supported for LINEAR and QUADRATIC models.
+
+  */
+  class OPENMS_DLLAPI TrafoModel
+  {
+  
+  private:
+    std::vector<double> coeff_;  //< Model coefficients (for both linear and quadratic models), estimated from the data
+    bool use_ppm_; //< during training, model is build on absolute or relative(ppm) predictions. predict(), i.e. applying the model, requires this information too
+    double rt_; //< retention time associated to the model (i.e. where the calibrant data was taken from)
+
+    static Math::RANSACParam* ransac_params_; //< global pointer, init to NULL at startup; set class-global RANSAC params
+    static double limit_offset_; //< acceptable boundary for the estimated offset; if estimated offset is larger (absolute) the model does not validate (isValidModel())
+    static double limit_scale_; //< acceptable boundary for the estimated scale; if estimated scale is larger (absolute) the model does not validate (isValidModel())
+    static double limit_power_; //< acceptable boundary for the estimated power; if estimated power is larger (absolute) the model does not validate (isValidModel())
+
+  public:
+
+    /**
+      @brief Default constructor
+    */
+    TrafoModel();
+
+    /**
+      @brief Default constructor
+
+      If you have external coefficients, use this constructor and the setCoefficients() method to
+      build a 'manual' model.
+      Afterwards, use applyTransformation() or predict() to calibrate your data.
+      If you call train(), the ppm-setting will be overwritten, depending on the type of training data.
+
+      @param ppm_model Are the coefficients derived from ppm calibration data, or from absolute deltas?
+    */
+    TrafoModel(bool ppm_model);
+
+    enum MODELTYPE { LINEAR, LINEAR_WEIGHTED, QUADRATIC, QUADRATIC_WEIGHTED, SIZE_OF_MODELTYPE };
+    static const std::string names_of_modeltype[]; //< strings corresponding to enum MODELTYPE
+    /**
+      @brief Convert string to enum
+
+      Returns 'SIZE_OF_MODELTYPE' if string is unknown.
+      @param name A string from names_of_modeltype[].
+      @return The corresponding enum value.
+    */
+    static MODELTYPE nameToEnum(const std::string& name);
+    /**
+      @brief Convert enum to string
+        
+      @param mt The enum value
+      @return Stringified version
+    */
+    static const std::string& enumToName(MODELTYPE mt); 
+
+
+    /**
+      @brief Set the global (program wide) parameters for RANSAC.
+
+      This is not done via member, to keep a small memory footprint since hundreds of
+      TrafoModels are expected to be build at the same time and the RANSAC params
+      should be identical for all of them.
+      
+      @param p RANSAC params
+    */
+    static void setRANSACParams(const Math::RANSACParam& p);
```

Comment:
is using a member a really problem? otherwise, I would probably prefer a smart pointer here to solve potential memory management issues.